packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/editingcontext/services/ResourceLoader.java

```diff
@@ -34,18 +45,34 @@
 @Service
 public class ResourceLoader implements IResourceLoader {
 
+    private static final MigrationData EMPTY_MIGRATION_DATA = new MigrationData("none", "0");
+
     private final Logger logger = LoggerFactory.getLogger(ResourceLoader.class);
 
+    private final List<IMigrationParticipant> migrationParticipants;
+
+    public ResourceLoader(List<IMigrationParticipant> migrationParticipants) {
+        this.migrationParticipants = migrationParticipants;
+    }
+
     @Override
     public Optional<Resource> toResource(ResourceSet resourceSet, String id, String name, String content) {
         Optional<Resource> optionalResource = Optional.empty();
 
+        var migrationData = getMigrationDataFromDocumentContent(content).orElse(EMPTY_MIGRATION_DATA);
+        var migrationExtendedMetaData = new MigrationExtendedMetaData(this.migrationParticipants, migrationData);
+        var migrationJsonProcessor = new MigrationJsonProcessor(this.migrationParticipants, migrationData);
+
+        HashMap<Object, Object> options = new HashMap<>();
+        options.put(JsonResource.OPTION_EXTENDED_META_DATA, migrationExtendedMetaData);
+        options.put(JsonResource.OPTION_JSON_RESSOURCE_PROCESSOR, migrationJsonProcessor);
+
         var resource = new JSONResourceFactory().createResourceFromPath(id);
+
         try (var inputStream = new ByteArrayInputStream(content.getBytes(StandardCharsets.UTF_8))) {
             resourceSet.getResources().add(resource);
-            resource.load(inputStream, null);
-
             resource.eAdapters().add(new ResourceMetadataAdapter(name));
+            resource.load(inputStream, options);
```

Comment:
That very little change may be necessary somewhere else too, I'll have to check

Reply:
You need to ensure that the adapter is setup everywhere it is used before the loading of the resource. This change, while understandable, impacts several locations.